src/MPAGSCipher/PlayfairCipher.cpp

```diff
@@ -0,0 +1,190 @@
+
+}
+
+void PlayfairCipher::setKey(const std::string& key) {
+
+    // store the original key
+    key_ = key;
+    
+    // Append the alphabet
+    key_ =+ "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+   
+    // Make sure the key is upper case
+    std::transform(std::begin(key_), std::end(key_), std::begin(key_), ::toupper);
+
+    // Remove non-alpha characters
+    auto nonalpha = [] (char str_char){
+        if (std::isalpha(str_char))
+            return false;
+        else
+            return true;
+    };
+
+    auto iter = std::remove_if(key_.begin(), key_.end(), nonalpha);
+    key_.erase(iter,key_.end());
+
+    // Change J -> I
+    auto jtoi = [] (char letter){
+        if (letter == 'J'){
+            letter = 'I';
+            return letter;
+        }
+        else {
+            return letter;
+        }
+
+    };
+
+    std::transform(key_.begin(), key_.end(), key_.begin(), jtoi);
+
+    // Remove duplicated letters
+    std::string l_encountered{""};
+
+    auto rem_dup = [&] (char l){
+        // check the char against a string of letters already found
+        std::size_t found = l_encountered.find(l);
+        // if char is in the list, mark for removal (return true)
+        if (found != std::string::npos){
+            return true;
+        // if char isn't already in list, add it to list and return false
+        }
+        else {
+            l_encountered += l;
+            return false;
+        }
+
+    };  
+
+    auto iter2 = std::remove_if(key_.begin(), key_.end(), rem_dup);
+    key_.erase(iter2,key_.end());
+
+    // Store the coords of each letter
+    // looping over each letter and storing coordinates and the letter in a map
+
+    // the position divided by 5 (as an integer) tells us the row number
+    // the position mod 5 tells us the column number
+   
+    for (std::size_t i{0}; i < keyLength_; i++){
+        std::size_t row_ind{i / gridSize_};
+        std::size_t col_ind{i % gridSize_};
+        
+        auto coords = std::make_pair(row_ind,col_ind);
+
+        ChartoCoordsMap_[key_[i]] = coords;
+        CoordstoCharMap_[coords] = key_[i];
+
+      
+    };
+
+    
+}
+
+std::string PlayfairCipher::applyCipher(const std::string& inputText, const CipherMode cipherMode) const {
+
+   
+
+    // Change J → I
+    auto jtoi = [] (char letter){
+        if (letter == 'J'){
+            letter = 'I';
+            return letter;
+        }
+        else {
+            return letter;
+        }
+
+    };
+
+    std::transform(inputText.begin(), inputText.end(), inputText.begin(), jtoi);
+
+    std::string examinedDigraphs{""};
+
+    // If repeated chars in a digraph add an X or Q if XX
+    // Look at each character in inputText
+    // If the character before is the same as the current one, replace with X + current character
+    // If both the characters are XX, replace with QX.
+    for (std::size_t i{0}; i < inputText.size(); i++){
+        // only doing something on every second letter
+        if (i == 1 % 2){
```

Comment:
I think you mean `if ( i % 2 == 1 )` here?

You can also achieve the same by starting at i=1 and incrementing the loop counter by 2:
```suggestion
    for (std::size_t i{1}; i < inputText.size(); i += 2){
```

However, if the supplied text was only a single character, this and your current implementation would ignore it. This is because it skips the `[0]` element and instead looks back to it when examining the `[1]` element, but if that does not exist nothing gets processed.  As such, it's better to act on the even-numbers and look forward to `[i+1]`:
```suggestion
    for (std::size_t i{0}; i < inputText.size(); i += 2){
```